### ISSUE TITLE
frustratingly, it is too easy to accidently click a link when scrolling on iphone

### ISSUE THREAD

**Issue by scd** [theCrag/website#1883]
From Guy...

Also still happy to play with the mobile version of the site, there are still some things that happen when scrolling on an iPhone and you hit the wrong button and a page loads. I get around this by scrolling on the right hand side but it's quite frustrating. Not sure if anyone else has mentioned this. This alone stopped me and a few friends from using the site on our mobiles

**Comment by brendanheywood** [comment]
@guyfi feel free to agree or comment more, but I think this boils down to too many links in list view inside a route, ie links to the route page, links to other mentioned routes, now links to the FA profile page. On mobile you probably only want a single interaction with the route as a whole which is to select it or deselect it. Once selected you'd want a list of actions, primarily 'tick', 'edit', 'add to wish list', 'see details' etc

There is probably also some dodgy scrolling handler logic if you are starting a scrolling on a topo.

So there is probably multiple overlapping issues that need to get sorted, if you can be patient and let me know which links you end up at the most that would help a lot to help me refine this.

**Comment by guyfi** [comment]
yes this is exactly the challenge, Way to many links makes it hard to use
without clicking the wrong thing.

even if just the tick button is used as a hyperlink.

for me i will only ever click the tick or click the checkbox for logging
multiple routes. other than that i scroll up and down to view the topo if
available.

On Sat, Mar 14, 2015 at 1:04 PM, Brendan Heywood <notifications@github.com>
wrote:

> @guyfi <https://github.com/guyfi> feel free to agree or comment more, but
> I think this boils down to too many links in list view inside a route, ie
> links to the route page, links to other mentioned routes, now links to the
> FA profile page. On mobile you probably only want a single interaction with
> the route as a whole which is to select it or deselect it. Once selected
> you'd want a list of actions, primarily 'tick', 'edit', 'add to wish list',
> 'see details' etc
>
> There is probably also some dodgy scrolling handler logic if you are
> starting a scrolling on a topo.
>
> So there is probably multiple overlapping issues that need to get sorted,
> if you can be patient and let me know which links you end up at the most
> that would help a lot to help me refine this.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/theCrag/website/issues/1883#issuecomment-79723964>.
>



-- 
Kind Regards,

Guy Moore
Phone : 0413 577 606
Email :djguyfi@gmail.com

**Comment by brendanheywood** [comment]
fyi I'm working on a whole new mobile approach to the list view. There are two changes I'm planning on which will make this better:

1) if you touch and then scroll if will never fire off a 'tap', this is the main bug right now (mind you it has gotten a lot better on iOS since apple made you tap twice when the hover style changes)

2) If you do tap on a route, then it only ever responds in the context of the route list item, not the actual bit you touched. Instead of going of to where-ever which could be the tick process, the route details, another route, or the ascents for that route, you'll instead get a modal with a list of those choices and a cancel button.

**Comment by ajlowndes** [comment]
Just to add to the above. Most people I know use thecrag as a guide-book, which means their priority is NOT to select individual routes ready to log, but to scroll through the area list and find what they want to climb.

Currently if you tap anywhere inside the bounding box of a route - i.e. the title, description or the whitespace around the description, the route will be selected, which is annoying and always requires me to scroll back up and find whatever route I accidentally tapped and un-select it. 
_Side note: I think your implementation of the "fastclick" library might be exacerbating the problem... if you are still using that._

For mobile, it would be better to make the tick-box on the left of the route name the ONLY way to select a route. 

Similarly, when scrolling past topos, it is easy to accidentally tap and select a route. To avoid that, I would only allow selection of individual routes IF the topo has been expanded to fill the screen. 

A further alternative if you didn't like any of the above - the "Log ascent" button at the bottom of the screen could be made into two buttons - the additional one being "Deselect all".

**Comment by brendanheywood** [comment]
Another thought that's been brewing for me for ages is that most of these links are the markdown auto linking to adjacent routes, eg "Blah blah stars 4m left of 'some route". In the majority of these cases if you click that link it should simply scroll up or down to that route, not take you to that other routes detail page.
